core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -509,20 +517,57 @@ class SpendReportingService(
         )
       }
 
-      query = getQuery(aggregations, spendExportConf)
-      queryJob = setUpQuery(query, spendExportConf, start, end, projectNames)
+      // look for a cached spend report
+      isCacheable = aggregationsAreCacheable(aggregations)
+      cachedResults <-
```

Comment:
I am finding this a bit hard to grok. I think the following pseudocode will make this easier
```
if (isCacheable) {
  val cachedResults = getCachedSpendReportData()
  if (cachedResults.isCacheValid) {
    cacheHit()
    return cachedResults
  } else {
    cacheMiss()
    populateCache()
    return getCachedSpendReportData()
  }
} else {
  return fetchDataFromBQ()
}
```
This approach requires that the cache is populated before returning results on a cache miss (which the current code does not guarantee)

Reply:
I was really trying to model the flow of this code to be parallel to the flow of the consolidated spend report code, to keep them aligned. I can see a few options here:
1. Restructure this code; leave the consolidated spend report alone.
2. Restructure neither; potentially refactor both in a follow-on PR.
3. Restructure both in this PR.

What do you think?

Reply:
The thing that bothers me the most is that a cache hit is a completely different code path than a cache miss. I think we should alway serve data from the cache. If it is not there, we put it in first. But it is always served from the cache. Then we avoid potential behavior differences or code duplication.

So my vote would be 3. But I tend to overload PRs

Reply:
Another option to not overload this PR too much could be to restructure this code and then restructure the consolidated spend report code in a follow-on PR. Unless there's some potential for code re-use by restructuring both?

Reply:
I took the approach of restructuring this code and leaving the consolidated spend report code as-is; trying to balance scope and code cleanliness